src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hosting/VSInternalServerCapabilitiesExtensions.cs

```diff
@@ -57,4 +61,33 @@ public static void EnableMapCodeProvider(this VSInternalServerCapabilities serve
     {
         serverCapabilities.MapCodeProvider = true;
     }
+
+    public static HashSet<string> HtmlAllowedAutoInsertTriggerCharacters { get; } = new(StringComparer.Ordinal) { "=", };
+    public static HashSet<string> CSharpAllowedAutoInsertTriggerCharacters { get; } = new(StringComparer.Ordinal) { "'", "/", "\n" };
+
+    public static void EnableOnAutoInsert(
+        this VSInternalServerCapabilities serverCapabilities,
+        bool singleServerSupport,
+        IEnumerable<string> triggerCharacters)
+    {
+        serverCapabilities.OnAutoInsertProvider = new VSInternalDocumentOnAutoInsertOptions()
+            .EnableOnAutoInsert(singleServerSupport, triggerCharacters);
+    }
+
+    public static VSInternalDocumentOnAutoInsertOptions EnableOnAutoInsert(
+        this VSInternalDocumentOnAutoInsertOptions options,
+        bool singleServerSupport,
+        IEnumerable<string> triggerCharacters)
+    {
+        if (singleServerSupport)
```

Comment:
Consider renaming this parameter so it communicates what it actually does. ie, `includeStandardTriggerCharacters` or something.

Reply:
Removed the parameter and moved code to merge in standard trigger chars to individual handlers per your suggestion